core/src/main/scala/caliban/schema/Schema.scala

```diff
@@ -683,3 +729,56 @@ trait TemporalSchema {
     )(formatter)
 
 }
+
+case class FieldType(isInput: Boolean, isSubscription: Boolean)
+
+case class PartiallyAppliedField[V](
+  name: String,
+  description: Option[String],
+  directives: List[Directive]
+) {
+  def apply[R, V1](fn: V => V1)(implicit ev: Schema[R, V1], ft: FieldType): (__Field, V => Step[R]) =
+    (
+      __Field(
+        name,
+        description,
+        Nil,
+        () =>
+          if (ev.optional) ev.toType_(ft.isInput, ft.isSubscription)
+          else Types.makeNonNull(ev.toType_(ft.isInput, ft.isSubscription)),
+        directives = Some(directives).filter(_.nonEmpty)
+      ),
+      (v: V) => ev.resolve(fn(v))
+    )
+}
+
+case class PartiallyAppliedFieldWithArgs[V, A](
+  name: String,
+  description: Option[String],
+  directives: List[Directive]
+) {
+  def apply[R, V1](
+    fn: V => V1
+  )(implicit ev1: Schema[R, V1], ev2: Schema[R, A], ft: FieldType): (__Field, V => Step[R]) = {
```

Comment:
If you take a `Schema[R, V1 => A]` maybe you can use it avoid duplicating that part from `functionSchema`?

Reply:
Not sure I understand how this would work, could you show an example of what you mean?

Reply:
Sorry I meant `A => V1`:
```scala
def apply[R, A, V1](
    fn: V => (A, V1)
  )(implicit ev: Schema[R, A => V1], ft: FieldType): (__Field, V => Step[R]) = {
  val args = ev.arguments
  ...
}
```

Reply:
Ah yes, nice